test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/InitializeTestProjectAttribute.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNetCore.Testing;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
+{
+    public class InitializeTestProjectAttribute : BeforeAfterTestAttribute
+    {
+        private readonly string _projectName;
+
+        public InitializeTestProjectAttribute(string projectName)
+        {
+            _projectName = projectName;
+        }
+
+        public override void Before(MethodInfo methodUnderTest)
+        {
+            if (!typeof(MSBuildIntegrationTestBase).GetTypeInfo().IsAssignableFrom(methodUnderTest.DeclaringType.GetTypeInfo()))
+            {
+                throw new InvalidOperationException($"This should be used on a class derived from {typeof(MSBuildIntegrationTestBase)}");
+            }
+
+            var solutionRoot = TestPathUtilities.GetSolutionRootDirectory("Razor");
+            if (solutionRoot == null)
+            {
+                throw new InvalidOperationException("Could not find solution root.");
+            }
+
+            var projectRoot = Path.Combine(solutionRoot, "test", "testapps", _projectName);
+            if (!Directory.Exists(projectRoot))
+            {
+                throw new InvalidOperationException($"Could not find project at '{projectRoot}'");
+            }
+
+            var tempDirectory = ProjectDirectory.Create();
+
+            try
+            {
+                CopyDirectory(new DirectoryInfo(projectRoot), new DirectoryInfo(tempDirectory.FilePath));
+
+                foreach (var project in Directory.EnumerateFiles(tempDirectory.FilePath, "*.csproj"))
```

Comment:
Do we expect to have more than one `csproj`?

Reply:
this seemed better than hardcoding the name.